### ISSUE TITLE
Not retaining invalid input for InputNumber creates an accessibility failure and usability issues

### ISSUE THREAD

**Issue by mellis481** [ant-design/ant-design#37925]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/format-tooltip-input-antd-4-23-5-forked-frqdwz?file=/demo.js)

##### Example for input out of range (1-10)
1. Put input in focus.
2. Type "11".
3. Hit Tab.

##### Example for invalid numeric input.
1. Put input in focus.
2. Type "asdf".
3. Hit Tab.

##### Example for input out of range (1-10)
Input sets value of 10.

##### Example for invalid number input.
Input clears.

### What is actually happening?

For both scenarios, the invalid value should remain in the input and the component should expose a property that indicates its invalid state.

| Environment | Info |
| --- | --- |
| antd | 4.23.5 |
| React | React |
| System | Windows 10 |
| Browser | All |

---
This is an accessibility failure of WCAG success criterion [3.3.1 Error Identification](https://www.w3.org/WAI/WCAG22/Understanding/error-identification.html) which requires invalid input to be effectively identified/communicated to users including clear instruction for remediation. Additionally, changing the invalid value to a different value without informing the user, creates a usability issue.

Lastly, this prevents InputNumber from having the same UX as other components (eg. Input) when used in a form. In other cases, invalid input is retained and validation can be configured to inform the user of input error and how to correct it. That's not possible with InputNumber when it's used in a form because of how InputNumber handles invalid input.
